src/main/java/org/radarcns/phone/PhoneLogManager.java

```diff
@@ -104,147 +112,239 @@ public PhoneLogManager(PhoneLogService phoneLogService, TableDataHandler dataHan
-        if (callLogReadFuture != null) {
-            callLogReadFuture.cancel(false);
-        }
+    public final synchronized void setCallAndSmsLogUpdateRate(final long period) {
+        // Create pending intent, which cancels currently active pending intent
+        PendingIntent pendingIntent = PendingIntent.getBroadcast(this.context, REQUEST_CODE_PENDING_INTENT,
+                new Intent(ACTIVITY_LAUNCH_WAKE), PendingIntent.FLAG_CANCEL_CURRENT);
+
+        // Get alarm manager and schedule it to run every period (seconds)
+        AlarmManager alarmManager = (AlarmManager) this.context.getSystemService(ALARM_SERVICE);
+        alarmManager.setInexactRepeating(AlarmManager.RTC_WAKEUP, System.currentTimeMillis(), period * 1000, pendingIntent);
 
-        callLogReadFuture = executor.scheduleAtFixedRate(new Runnable() {
+        // Activity to perform when alarm is triggered
+        this.context.registerReceiver(new BroadcastReceiver() {
             @Override
-            public void run() {
-
-                try {
-                    final long initialDateRead = Long.parseLong(storage.getOrSet(LAST_CALL_KEY, "0"));
-                    long lastDateRead = initialDateRead;
-                    long threshold = System.currentTimeMillis() / 1000L - period;
-                    if (lastDateRead < threshold) {
-                        lastDateRead = threshold;
-                    }
-
-                    try (Cursor c = getService().getContentResolver().query(CallLog.Calls.CONTENT_URI, null, CallLog.Calls.DATE + " > " + lastDateRead, null, CallLog.Calls.DATE + " ASC")) {
-                        if (c == null) {
-                            return;
-                        }
-
-                        while (c.moveToNext()) {
-                            long date = c.getLong(c.getColumnIndex(CallLog.Calls.DATE));
-
-                            processCall(date / 1000d,
-                                    c.getString(c.getColumnIndex(CallLog.Calls.NUMBER)),
-                                    c.getFloat(c.getColumnIndex(CallLog.Calls.DURATION)),
-                                    c.getInt(c.getColumnIndex(CallLog.Calls.TYPE)));
-                            lastDateRead = date;
-                        }
-                    } catch (Throwable t) {
-                        logger.warn("Error in processing the call log: {}", t.getMessage());
-                        t.printStackTrace();
-                    } finally {
-                        if (lastDateRead != initialDateRead) {
-                            storage.put(LAST_CALL_KEY, Long.toString(lastDateRead));
-                        }
-                    }
-                } catch (IOException ex) {
-                    logger.error("Failed to read or write last call processed.", ex);
-                }
+            public void onReceive(Context context, Intent intent) {
+                processCallLog();
```

Comment:
The method does not get called when application is not running.

Test setup: 60 second alarm interval, started app and closed it (both foreground and background service). Send sms to phone to make sure there was something to process and waited 3 min.
Nothing happened.
New sms was processed when reopening app.